src/Controls/src/Core/NavigationPage/NavigationPage.cs

```diff
@@ -699,8 +703,7 @@ private protected override void OnHandlerChangedCore()
 				},
 				() =>
 				{
-					// TODO this is the wrong navigation type
```

Comment:
This was passing the wrong navigation type parameter. If we not decide to update the navigation EventArgs with the properties included in this PR, the changes to fix this should be included in another one.